src/parallax/useIntersectionObserver.ts

```diff
@@ -0,0 +1,32 @@
+import { RefObject, useEffect, useState } from "react";
+
+type Options = {
+  root: HTMLElement | null;
+  rootMargin: string;
+  threshold: number;
+};
+
+export function useIntersectionObserver<T>(
+  ref: RefObject<T>,
+  observerOptions: Options
+) {
+  const [isInViewport, setIsInViewPort] = useState(false);
+  const [isFirstShown, setIsFirstShown] = useState(false);
```

Comment:
I remembered a better name `hasShown` ;)

Reply:
I like it hehe, changed